src/coreclr/jit/codegenarm64.cpp

```diff
@@ -335,10 +335,14 @@ bool CodeGen::genInstrWithConstant(instruction ins,
             break;
 
         case INS_ldrsb:
-        case INS_ldrsh:
-        case INS_ldrsw:
         case INS_ldrb:
+            immFitsInIns = emitter::emitIns_valid_imm_for_ldst_offset(imm, EA_1BYTE);
```

Comment:
Why is `size` not already EA_1BYTE in this case?

Reply:
It is passed by the caller `genEnregisterOSRArgsAndLocals()` using `emitActualTypeSize(lclTyp)`, which for `TYP_BYTE` returns `EA_4BYTE`.

Reply:
It seems like the bug is in `genEnregisterOSRArgsAndLocals()` -- it appears lclTyp is TYP_BYTE so it is asking for a ldrb but it passes the actual size (4) instead of size 1. It looks like lclTyp can only be small for OSR local struct fields. Maybe it should use `const emitAttr  size = emitTypeSize(lclTyp);` instead of using `emitActualTypeSize()`? @AndyAyersMS ?

Reply:
Yes, I think we should be using `emitTypeSize` in `genEnregisterOSRArgsAndLocals`.

Reply:
> It looks like lclTyp can only be small for OSR local struct fields.

Why is the reasoning behind it?

> `emitTypeSize`

I never understood the difference between `emitTypeSize` and `emitActualTypeSize()`. In which scenarios should we use one over other?

Reply:
Turns out that for `emit_R_R_I()` format, if we are sign extending, we expect the size to be valid general size i.e. 4 bytes or 8 bytes, so changing to `emitTypeSize()` hits an assert. I will follow up offline with @BruceForstall .